tests/test_surface_tally/test_surface_tally.py

```diff
@@ -0,0 +1,164 @@
+        borated_water.add_element('O', 2.4e-2)
+        borated_water.add_s_alpha_beta('c_H_in_H2O')
+
+        # Instantiate a Materials collection and export to XML
+        materials_file = openmc.Materials([uo2, borated_water])
+
+        materials_file.export_to_xml()
+
+        # Instantiate ZCylinder surfaces
+        fuel_or = openmc.ZCylinder(surface_id=1, x0=0, y0=0, R=0.4, name='Fuel OR')
+        left    = openmc.XPlane(surface_id=2, x0=-0.62992, name='left')
+        right   = openmc.XPlane(surface_id=3, x0=0.62992, name='right')
+        bottom  = openmc.YPlane(y0=-0.62992, name='bottom')
+        top     = openmc.YPlane(y0=0.62992, name='top')
+
+        left.boundary_type = 'vacuum'
+        right.boundary_type = 'reflective'
+        top.boundary_type = 'reflective'
+        bottom.boundary_type = 'reflective'
+
+        # Instantiate Cells
+        fuel = openmc.Cell(name='fuel')
+        water = openmc.Cell(name='water')
+
+        # Use surface half-spaces to define regions
+        fuel.region = -fuel_or
+        water.region = +fuel_or & -right & +bottom & -top
+
+        # Register Materials with Cells
+        fuel.fill  = uo2
+        water.fill = borated_water
+
+        # Instantiate pin cell Universe
+        pin_cell = openmc.Universe(name='pin cell')
+        pin_cell.add_cells([fuel, water])
+
+        # Instantiate root Cell and Universe
+        root_cell = openmc.Cell(name='root cell')
+        root_cell.region = +left & -right & +bottom & -top
+        root_cell.fill = pin_cell
+        root_univ = openmc.Universe(universe_id=0, name='root universe')
+        root_univ.add_cell(root_cell)
+
+        # Instantiate a Geometry, register the root Universe, and export to XML
+        geometry = openmc.Geometry(root_univ)
+        geometry.export_to_xml()
+
+        # Instantiate a Settings object, set all runtime parameters, and export to XML
+        settings_file = openmc.Settings()
+        settings_file.batches = 10
+        settings_file.inactive = 0
+        settings_file.particles = 1000
+
+        # Create an initial uniform spatial source distribution over fissionable zones
+        bounds = [-0.62992, -0.62992, -1, 0.62992, 0.62992, 1]
+        uniform_dist = openmc.stats.Box(bounds[:3], bounds[3:], only_fissionable=True)
+        settings_file.source = openmc.source.Source(space=uniform_dist)
+        settings_file.export_to_xml()
+
+        # Tallies file
+        tallies_file = openmc.Tallies()
+
+        # Cell to cell tallies
+        # These filters are same for all tallies
+        two_groups = np.array([0., 4, 20.]) * 1e6
+        energy_filter     = openmc.EnergyFilter(two_groups)
+        polar_filter      = openmc.PolarFilter([0, np.pi / 4, np.pi])
+        azimuthal_filter  = openmc.AzimuthalFilter([0, np.pi / 4, np.pi])
+
+
+        cell_to_cell_tallies = []
+        tally_index = 0 
+
+        for cell1 in pin_cell.get_all_cells():
+            for cell2 in pin_cell.get_all_cells():
```

Comment:
This double-loop seems overly complicated since there are only two cells in the `pin_cell` universe. You might as well just manually add the filters/tallies needed.